### ISSUE TITLE
Refactor Cloud ML Examples and integrate with deployment docs

### ISSUE THREAD

**Issue by jacobtomlinson** [rapidsai/cloud-ml-examples#197]
**TLDR** This repository holds loads of really useful example notebooks and documentation on running RAPIDS ML workflows on the cloud but many examples are old or out of date and this repo is hard to maintain. I'd like to refactor this repo and integrate it with the deployment docs to fix this.

### Challenges with maintaining this repo

- There is no index/toc and maintaining one is hard.
-  Cloud deployment instructions get duplicated between notebooks. 
   - These instructions go stale and it's hard to update them all.
   - We want https://docs.rapids.ai/deployment/stable/ to be the one source of truth for this.
- Notebooks make opinionated choices about cloud/platform/tool/workflow. 
  - We might have a GCP + Kubernetes + Triton FIL + Mortgage example, but not the same for AWS/Azure/etc or using ECS/VMs/etc. 
  - We don't want to create a notebook for every option on every axis, it would be hard to keep them up to date.
- Examples are often written with a specific audience or customer in mind and then abandoned once they have served their purpose.

### Proposal

I propose that we integrate this repo more tightly with the Deployment documentation repo to try and solve these problems.

- Move as much cloud setup documentation out of notebooks as possible and cross-link to deployment docs pages.
- Use tagging to allow easy navigation between vendor/platform/tool specific reference docs and opinionated workflow examples.
- Build notebooks as a Sphinx site to allow easier navigation and discovery. Either by merging this repo into the deployment one or by setting up some build automation.

#### Building examples as a website

We could use [myst-nb](https://myst-nb.readthedocs.io/en/latest/) to build the notebooks into markdown files that get synced into the deployment docs repo. They would then get rendered into an examples section of the deployment docs.

![Untitled-2022-11-23-1045 excalidraw](https://user-images.githubusercontent.com/1610850/203536346-c3e87851-3aab-400a-a7d8-1d62be6685d9.png)

The top levels example page could just be a full list of all the notebooks but filterable by tag so that every notebook is discoverable and folks can easily find relevant examples to them.

![Untitled-2022-11-23-1045 excalidraw(3)](https://user-images.githubusercontent.com/1610850/203537511-69808dc4-8a3f-4b68-9baf-888b5002cecb.png)

We could then use those tags to show relevant examples at the bottom of each deployment documentation page.

![Untitled-2022-11-23-1045 excalidraw(2)](https://user-images.githubusercontent.com/1610850/203536847-bbf2fdd6-e4e5-48c8-83f0-eb411e9de330.png)
